src/Integration.Vsix.UnitTests/Commands/ProjectExcludePropertyToggleCommandTests.cs

```diff
@@ -0,0 +1,355 @@
+            this.projectSystem.SelectedProjects = new[] { project };
+
+            // Test case 1: true --toggle--> clears property
+            this.SetExcludeProperty(project, true);
+
+            // Act
+            testSubject.Invoke(command, null);
+
+            // Verify
+            this.VerifyExcludeProperty(project, null);
+
+            // Test case 2: no property --toggle--> true
+            this.SetExcludeProperty(project, null);
+
+            // Act
+            testSubject.Invoke(command, null);
+
+            // Verify
+            this.VerifyExcludeProperty(project, true);
+        }
+
+        [TestMethod]
+        public void ProjectExcludePropertyToggleCommand_Invoke_MultipleProjects_ConsistentPropValues_TogglesValues()
+        {
+            // Setup
+            OleMenuCommand command = CommandHelper.CreateRandomOleMenuCommand();
+            command.Enabled = true;
+
+            var testSubject = new ProjectExcludePropertyToggleCommand(this.serviceProvider);
+
+            var p1 = new ProjectMock("good1.proj");
+            var p2 = new ProjectMock("good2.proj");
+            p1.SetCSProjectKind();
+            p2.SetCSProjectKind();
+            this.projectSystem.SelectedProjects = new[] { p1, p2 };
+
+            // Test case 1: all not set --toggle--> all true
+            // Act
+            testSubject.Invoke(command, null);
+
+            // Verify
+            this.VerifyExcludeProperty(p1, true);
+            this.VerifyExcludeProperty(p2, true);
+
+            // Test case 2: all true --toggle--> all not set
+            // Setup
+            this.SetExcludeProperty(p1, true);
+            this.SetExcludeProperty(p2, true);
+
+            // Act
+            testSubject.Invoke(command, null);
+
+            // Verify
+            this.VerifyExcludeProperty(p1, null);
+            this.VerifyExcludeProperty(p2, null);
+        }
+
+        [TestMethod]
+        public void ProjectExcludePropertyToggleCommand_Invoke_MultipleProjects_MixedPropValues_SetIsExcludedTrue()
+        {
+            // Setup
+            OleMenuCommand command = CommandHelper.CreateRandomOleMenuCommand();
+            command.Enabled = true;
+
+            var testSubject = new ProjectExcludePropertyToggleCommand(this.serviceProvider);
+
+            var p1 = new ProjectMock("good1.proj");
+            var p2 = new ProjectMock("good2.proj");
+            p1.SetCSProjectKind();
+            p2.SetCSProjectKind();
+            this.projectSystem.SelectedProjects = new[] { p1, p2 };
+
+            this.SetExcludeProperty(p1, true);
+            this.SetExcludeProperty(p2, null);
+
+            // Act
+            testSubject.Invoke(command, null);
+
+            // Verify
+            this.VerifyExcludeProperty(p1, true);
+            this.VerifyExcludeProperty(p2, true);
+        }
+
+        [TestMethod]
+        public void ProjectExcludePropertyToggleCommand_QueryStatus_SingleProject_Supported_IsEnabledIsVisible()
```

Comment:
SupportedProject rather than Supported since commands had that flag and I didn't understand why would you check that flag.